StripeConnect/StripeConnectTests/Internal/Webview/ConnectComponentWebViewControllerTests.swift

```diff
@@ -0,0 +1,259 @@
+                                                      didFailLoadWithError: { error = $0 })
+        // Mock that loading indicator is animating
+        webVC.activityIndicator.startAnimating()
+        webVC.webView.navigationDelegate?.webView?(webVC.webView, didFailProvisionalNavigation: nil, withError: NSError(domain: "test_domain", code: 111))
+        XCTAssertEqual((error as NSError?)?.domain, "test_domain")
+        XCTAssertEqual((error as NSError?)?.code, 111)
+    }
+
+    func testDidFailNavigationTriggersLoadError() {
+        var error: Error?
+        let componentManager = componentManagerAssertingOnFetch()
+        let webVC = ConnectComponentWebViewController(componentManager: componentManager,
+                                                      componentType: .payouts,
+                                                      loadContent: false,
+                                                      didFailLoadWithError: { error = $0 })
+        // Mock that loading indicator is animating
+        webVC.activityIndicator.startAnimating()
+        webVC.webView.navigationDelegate?.webView?(webVC.webView, didFail: nil, withError: NSError(domain: "test_domain", code: 111))
+        XCTAssertEqual((error as NSError?)?.domain, "test_domain")
+        XCTAssertEqual((error as NSError?)?.code, 111)
+        // Loading indicator should stop
+        XCTAssertFalse(webVC.activityIndicator.isAnimating)
+    }
+
+    @MainActor
+    func testDidReceiveNon200StatusTriggersLoadError() async {
+        var error: Error?
+        let componentManager = componentManagerAssertingOnFetch()
+        let webVC = ConnectComponentWebViewController(componentManager: componentManager,
+                                                      componentType: .payouts,
+                                                      loadContent: false,
+                                                      didFailLoadWithError: { error = $0 })
+        _ = await webVC.webView(webVC.webView, decidePolicyFor: MockNavigationResponse(response: HTTPURLResponse(url: URL(string: "https://stripe.com")!, statusCode: 404, httpVersion: nil, headerFields: nil)!))
+        XCTAssertEqual((error as? HTTPStatusError)?.errorCode, 404)
+        // Loading indicator should stop
+        XCTAssertFalse(webVC.activityIndicator.isAnimating)
+    }
+}
+// MARK: - Helpers
+
+private extension ConnectComponentWebViewControllerTests {
+    func componentManagerAssertingOnFetch(appearance: Appearance = .default, fonts: [EmbeddedComponentManager.CustomFontSource] = []) -> EmbeddedComponentManager {
+        EmbeddedComponentManager(apiClient: .init(publishableKey: "test"),
+                                 appearance: appearance,
+                                 fonts: fonts,
+                                 fetchClientSecret: {
+            XCTFail("Client secret should not be retrieved in this test")
+            return ""
+        })
+    }
```

Comment:
Moved `componentManagerAssertingOnFetch` into an extension since it was sandwiched between multiple test functions